packages/clay-core/src/vertical-bar/Resizer.tsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * SPDX-FileCopyrightText: © 2023 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {Keys} from '@clayui/shared';
+import React, {useContext} from 'react';
+
+import {VerticalBarContext} from './context';
+
+type Props = {
+	/**
+	 * Sets the CSS className for the component.
+	 */
+	className?: string;
+
+	/**
+	 * Reference to Panel
+	 */
+	nodeRef?: React.RefObject<HTMLDivElement>;
+};
+
+let keyDownCounter = 0;
+
+export function Resizer({nodeRef, ...otherProps}: Props) {
+	const {
+		internalPanelWidth,
+		onPanelWidthChange,
+		panelWidthMax,
+		panelWidthMin,
+		position,
+	} = useContext(VerticalBarContext);
+
+	const positionLeft = position === 'left';
+
+	const getStartWidth = () => {
+		return nodeRef?.current?.offsetWidth || 320;
+	};
+
+	const decreasePanelWidth = (delta = 1, startWidth = getStartWidth()) => {
+		const width =
+			internalPanelWidth - delta < panelWidthMin
+				? panelWidthMin
+				: startWidth - delta;
+
+		onPanelWidthChange(Math.round(width));
+	};
+
+	const increasePanelWidth = (delta = 1, startWidth = getStartWidth()) => {
+		const width =
+			internalPanelWidth + delta > panelWidthMax
+				? panelWidthMax
+				: startWidth + delta;
+
+		onPanelWidthChange(Math.round(width));
+	};
+
+	return (
+		<div
+			{...otherProps}
+			onKeyDown={(event: React.KeyboardEvent<HTMLDivElement>) => {
+				const delta = keyDownCounter > 7 ? 10 : 1;
+
+				keyDownCounter++;
+
+				switch (event.key) {
+					case Keys.Down: {
+						decreasePanelWidth(delta);
+
+						break;
+					}
+					case Keys.Left: {
+						if (positionLeft) {
+							decreasePanelWidth(delta);
+						} else {
+							increasePanelWidth(delta);
+						}
+
+						break;
+					}
+					case Keys.Right: {
+						if (positionLeft) {
+							increasePanelWidth(delta);
+						} else {
+							decreasePanelWidth(delta);
+						}
+
+						break;
+					}
+					case Keys.Up: {
+						increasePanelWidth(delta);
+
+						break;
+					}
+					default: {
+						break;
+					}
+				}
+			}}
+			onKeyUp={() => {
+				keyDownCounter = 0;
+			}}
+			onPointerDown={(event) => {
+				const startWidth = getStartWidth();
+				const startXPos = event.pageX;
+
+				function onResizerMove(event: any) {
+					const delta = Math.abs(event.pageX - startXPos);
+
+					if (
+						(event.pageX >= startXPos && positionLeft) ||
+						(event.pageX < startXPos && !positionLeft)
+					) {
+						increasePanelWidth(delta, startWidth);
+					} else if (
+						(event.pageX < startXPos && positionLeft) ||
+						(event.pageX >= startXPos && !positionLeft)
+					) {
+						decreasePanelWidth(delta, startWidth);
+					}
+				}
+
+				function removeResizerEvents() {
+					document.removeEventListener('pointermove', onResizerMove);
+
+					document.removeEventListener(
+						'pointerup',
+						removeResizerEvents
+					);
+				}
+
+				if (event.button === 0) {
```

Comment:
What specific case is this?

Reply:
Right clicking the resizer doesn't remove the pointer events. After you right click and go back to the page, the resizer is stuck to cursor and you have to click it again to unstick it. `event.button === 0` makes sure to add the listeners only when the main mouse button is clicked.

Reply:
Maybe 0 should be declared as a named constant to avoid having a magic number here. IE: `const MAIN_BUTTON = 0;`

Reply:
I see, makes sense, maybe another strategy to insure it is removed is to stabilize the listener functions with `useCallback` and add a useEffect with zero dependency and a callback function that cleans up the methods, so when the component is unmounted we always make sure we are removing.